iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/unique_ptr.inl

```diff
@@ -25,148 +25,111 @@ namespace iox
 namespace cxx
 {
 template <typename T>
-unique_ptr<T>::unique_ptr(T* const ptr, const function<void(T*)>& deleter) noexcept
-    : m_ptr(ptr)
+inline unique_ptr<T>::unique_ptr(T* const object, const function<void(T*)>& deleter) noexcept
+    : m_ptr(object)
```

Comment:
For consistency reasons I would rename the parameter to `ptr`, but no changes required now.